frontend/src/lib/components/common/confirmationModal/UnsavedConfirmationModal.svelte

```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
+	import ConfirmationModal from './ConfirmationModal.svelte'
+	import { beforeNavigate, goto } from '$app/navigation'
+	import { onDestroy } from 'svelte/internal'
+	import { dirtyStore } from './dirtyStore'
+
+	let navigationState: { from: URL; to: URL | null; cancel: () => void } | undefined = undefined
+	$: open = Boolean(navigationState)
+
+	beforeNavigate((newNavigationState) => {
+		if (
+			!navigationState &&
+			$dirtyStore &&
+			newNavigationState.to?.pathname !== newNavigationState.from.pathname
```

Comment:
We only cancel routing when from pathname is different from to pathname: In the script/flow builder, the pathname stays the same but the "step" query parameters changes

Reply:
very clever!